### ISSUE TITLE
Azure, `make validate` does not work from localbox

### ISSUE THREAD

**Issue by Rushit** [kubernetes/kubernetes-anywhere#260]
kubernetes-anywhere (master) $ `make validate` 
KUBECONFIG="$(pwd)/phase1/$(jq -r '.phase1.cloud_provider' .config.json)/.tmp/kubeconfig.json" ./util/validate
Validation: Expected 5 healthy nodes; found        0. (10s elapsed)
Validation: Expected 5 healthy nodes; found        0. (20s elapsed)
Validation: Expected 5 healthy nodes; found        0. (30s elapsed)
Validation: Expected 5 healthy nodes; found        0. (40s elapsed)
Validation: Expected 5 healthy nodes; found        0. (50s elapsed)
Validation: Expected 5 healthy nodes; found        0. (60s elapsed)
Validation: Expected 5 healthy nodes; found        0. (70s elapsed)

`make deploy` creates the cluster. I can see into azure console, but I cannot run kubectl from local too. Is there any step I missed? or anything related to opening up port or something on azure side?

**Comment by colemickens** [comment]
No, there shouldn't be any additional steps needed. How long did you wait? Can you give it another minute or two?

If it doesn't improve... If you're willing, can you ssh to your master node (the credentials are in `<kube-anywhere>/.config.json`) and see if you can diagnose anything?

If you can SSH to the master node, please execute:
* `sudo journalctl -u kubelet 2>&1 | nc termbin.com 9999`
* `sudo cat /var/log/cloud-init-output.log | nc termbin.com 9999`

Each of those will output a link, reply with them here and I can take a look.

**Comment by colemickens** [comment]
Are you still experiencing issues here?

**Comment by 486** [comment]
Hi, I have the exact same problem. `make deploy` went through but then it stalls:

```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

smartkube-master-pip = [redacted]
make[1]: Leaving directory '/opt/kubernetes-anywhere'
KUBECONFIG="$(pwd)/phase1/$(jq -r '.phase1.cloud_provider' .config.json)/.tmp/kubeconfig.json" ./util/validate
Validation: Expected 5 healthy nodes; found 0. (10s elapsed)
Validation: Expected 5 healthy nodes; found 0. (20s elapsed)
Validation: Expected 5 healthy nodes; found 0. (30s elapsed)
Validation: Expected 5 healthy nodes; found 0. (40s elapsed)

```

I should mention that the output is from a consecutive run of `make deploy`. I can see that the Azure resource group contains a master and 4 node VMs, an availability set, nics and other stuff. The group had been up for several hours when I ran the command again.

I had to call `deploy` multiple times because it failed with messages like 

```
Error refreshing state: 2 error(s) occurred:

* stream closed
* connection is shut down
Error configuring: 2 error(s) occurred:

* connection write timeout
* Error configuring template: EOF
```

**Comment by colemickens** [comment]
Those errors look like they came from Terraform, but it's not enough information for me to go on.

If your ServicePrincipal credentials are wrong, the cluster won't come up properly. Did you specify them yourself or did you let it generate it for you?

If you're unsure, I'd recommend following the troubleshooting steps I listed just above. That should give a definitive answer to what is going on.

**Comment by 486** [comment]
I let it generate them.
In the meantime I removed the whole resource group. I tried it again from scratch and it seems to have worked this time. 
For this run I used a ssh password without special characters and didn't change the VM sizes. Also I could complete everything in one go - didn't work the last time.

As a sidenote, one thing that kept me from being successful on my first attempt was that I didn't know the datacenter region codes of azure:

```
centralus,eastasia,southeastasia,eastus,eastus2,westus,westus2,northcentralus,southcentralus,westcentralus,northeurope,westeurope,japaneast,japanwest,brazilsouth,australiasoutheast,australiaeast,westindia,southindia,centralindia,canadacentral,canadaeast,uksouth,ukwest
```

I didn't find them on Google so I guessed, and of course guessed wrong. I corrected it later but maybe this broke it in the first place.
